amip_interp/amip_interp.F90

```diff
@@ -265,31 +280,59 @@ module amip_interp_mod
 !!
 !> @ingroup amip_interp_mod
 interface amip_interp_new
-   module procedure amip_interp_new_1d
-   module procedure amip_interp_new_2d
+   module procedure amip_interp_new_1d_r4, amip_interp_new_1d_r8
+   module procedure amip_interp_new_2d_r4, amip_interp_new_2d_r8
+end interface
+
+!> Initialize amip_interp. To explicitly initialize the module in r4 mode or r8 mode,
+!! call either amip_interp_init_r4 or amip_interp_init_r8. amip_interp_init is an
+!! alias for amip_interp_init_r8.
+interface amip_interp_init
+  module procedure amip_interp_init_r8
 end interface
 
+!> Frees data associated with a amip_interp_type variable. Should be used for any
+!! variables initialized via @ref amip_interp_new.
+!> @param[inout] Interp A defined data type variable initialized by amip_interp_new and used
+!! when calling get_amip_sst and get_amip_ice.
+interface amip_interp_del
+  module procedure amip_interp_del_r4, amip_interp_del_r8
+end interface amip_interp_del
 
 !----- public data type ------
 
-!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers data.
+!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers
+!! data (r4_kind flavor).
 !> @ingroup amip_interp_mod
-type amip_interp_type
+type amip_interp_type_r4
```

Comment:
Sorry didn't realize amp_interp_type was public.  Is there anyway to mixed precision to amip_interp without changing the name of amip_interp_type?

Reply:
This could be done by making `data1` and `data2` both `class(*)` and wrapping all code that uses them in `select type` blocks.

Reply:
`amip_interp_type` has been brought back in fb78dae60417e134dcdc2c8f2ac3181ab5fed580.